app/controlplane/plugins/core/guac/v1/guac.go

```diff
@@ -0,0 +1,268 @@
+//
+// Copyright 2023 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guac
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"cloud.google.com/go/storage"
+	"code.cloudfoundry.org/bytefmt"
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/app/controlplane/plugins/sdk/v1"
+	"github.com/go-kratos/kratos/v2/log"
+	"google.golang.org/api/option"
+)
+
+// Integration implements of a FanOut integration
+// See https://github.com/chainloop-dev/chainloop/blob/main/app/controlplane/plugins/README.md for more information
+type Integration struct {
+	*sdk.FanOutIntegration
+}
+
+const providerGCS = "gcs"
+
+// Options provided to the user when registering the integration
+type registrationRequest struct {
+	Provider    string `json:"provider,omitempty" jsonschema:"minLength=1,description=Blob storage provider: default gcs,enum=gcs"`
+	Bucket      string `json:"bucket" jsonschema:"minLength=1,description=Bucket name where to store the artifacts"`
+	Credentials string `json:"credentials" jsonschema:"minLength=2,description=Credentials to access the bucket"`
+}
+
+// No customization options are supported during attachment time
+type attachmentRequest struct{}
+
+// State stored to be retrieved later on during the execution of the actual dispatch
+// NOTE: the credentials are not stored in this state but instead on a secure location, see registrationResponse below
+type registrationState struct {
+	Bucket   string `json:"bucket"`
+	Provider string `json:"provider"`
+}
+
+func New(l log.Logger) (sdk.FanOut, error) {
+	base, err := sdk.NewFanOut(
+		&sdk.NewParams{
+			ID:          "guac",
+			Version:     "1.0",
+			Description: "Export Attestation and SBOMs metadata to a blob storage backend so guacsec/guac can consume it",
+			Logger:      l,
+			InputSchema: &sdk.InputSchema{
+				Registration: registrationRequest{},
+				Attachment:   attachmentRequest{},
+			},
+		},
+		// This plugin subscribes to SBOMs and attestations (note: attestations come by default)
+		// In the future the kinds will expand to match
+		// https://github.com/guacsec/guac#supported-input-formats
```

Comment:
What if I push to GCS something what is not supported by guac? Will it be ignored? Will it make it fail?

Reply:
For what I can see their parser will fail and ignore the file. In fact that's what will happen with our attestation today since we have a custom in-toto predicate.

But I'd say that's not our scope, we are sending content that's technically supported by Guac, and that system should be smart enough to pick up the pieces that understand and ignore the rest. 

Otherwise we'll not be able to cater multiple versions of Guac, i.e Guac that supports newer versions or even new material types.

WDYT?

Reply:
I am not sure. Let's keep it as it is for now and then work on PR to have our chainloop in-toto predicate supported in Guac?

Reply:
Should we add somewhere the version of Guac supported by this version of our plugin?

Reply:
Done in the readme, v0.1 beta+